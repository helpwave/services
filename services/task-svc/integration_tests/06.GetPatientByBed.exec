#!/bin/bash

echo "Exists, with patient:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"bed_id": "31d2cb94-c6fd-4d98-acd4-128c31d51693"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientByBed

echo "Exists, no Patient:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"bed_id": "018bd012-4aa1-468f-b57c-c0028507c332"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientByBed 2>&1 || exit 0

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"bed_id": "32d2cb94-c6fd-4d98-acd4-128c31d51693"}' \
	localhost:3001 \
	proto.services.task_svc.v1.PatientService/GetPatientByBed 2>&1 || exit 0
