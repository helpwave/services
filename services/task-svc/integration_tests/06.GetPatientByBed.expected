Exists, with patient:
{
  "id": "2a7f62d2-926c-4a0b-a963-2b4cdc89e1e7",
  "humanReadableIdentifier": "Prudy Shimoni",
  "notes": "Contusion of right little finger with damage to nail",
  "bedId": "31d2cb94-c6fd-4d98-acd4-128c31d51693"
}
Exists, no Patient:
ERROR:
  Code: InvalidArgument
  Message: id not found
